.github/scripts/delete-old-cloudsmith-artifacts.sh

```diff
@@ -1,42 +1,74 @@
 #!/bin/bash
 # Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
-dnf install -y 'dnf-command(config-manager)' jq
 
-set -euo pipefail
+if [ "${RUNNER_DEBUG:-}" == "1" ]; then
+    set -o xtrace
+fi
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+dnf install -yq 'dnf-command(config-manager)' jq
+
 MAX_NUMBER_OF_RELEASES=32
 
 # Cloudsmith repo
 rpm --import 'https://dl.cloudsmith.io/public/vespa/open-source-rpms/gpg.0F3DA3C70D35DA7B.key'
-curl -1sLf 'https://dl.cloudsmith.io/public/vespa/open-source-rpms/config.rpm.txt?distro=el&codename=8' > /tmp/vespa-open-source-rpms.repo
+curl  --silent --show-error --location --fail \
+  --tlsv1 --output /tmp/vespa-open-source-rpms.repo \
+  'https://dl.cloudsmith.io/public/vespa/open-source-rpms/config.rpm.txt?distro=el&codename=8'
 dnf config-manager --add-repo '/tmp/vespa-open-source-rpms.repo'
 rm -f /tmp/vespa-open-source-rpms.repo
 
-# Allow the last Vespa 7 release to remain in the repo
-VERSIONS_TO_DELETE=$(dnf list -y --quiet --showduplicates --disablerepo='*' --enablerepo=vespa-open-source-rpms vespa | awk '/[0-9].*\.[0-9].*\.[0-9].*/{print $2}' | sort -V | grep -v "7.594.36" | head -n -$MAX_NUMBER_OF_RELEASES)
+dnf list -y --quiet --showduplicates --disablerepo='*' --enablerepo=vespa-open-source-rpms vespa >/tmp/dnf-stdout 2>/tmp/dnf-stderr || true
```

Comment:
By testing for failure I was thinking of testing the exit status, so something like

```
if DNF_STDOUT=$(dnf list -y --quiet --showduplicates); then
  ...
else
  ... failure, be sure to exit with some message, whether something was written to stderr or not
fi
```

Usually, a failing command writes something to stderr, so it is equivalent.

Reply:
I believe your solution is better because `dnf` does not seem to care about the `--quiet`  option and _only the first run_ it writes _"Importing GPG key 0x0D35DA7B"_ to stderr... 😒

Reply:
I solved it by adding `dnf makecache` before, so that it's pre-cached. 🙃